packages/amplication-git-utils/src/git/git.service.ts

```diff
@@ -89,39 +99,213 @@ export class GitClientService {
+        owner,
+        repositoryName,
+      });
+
+      const cloneDir = normalize(
+        join(cloneFolder, this.provider.name, owner, repositoryName, randomUUID)
+      );
+
+      await gitClient.clone(cloneUrl, cloneDir);
+
+      await this.restoreAmplicationBranchIfNotExists({
         owner,
         repositoryName,
         branchName,
-      }),
-        await this.provider.createCommit({
+        gitClient,
+      });
+
+      const diffFolder = normalize(
+        join(
+          `.amplication/diffs`,
+          this.provider.name,
           owner,
+
           repositoryName,
-          commitMessage,
-          branchName,
-          files: preparedFiles,
+          randomUUID
+        )
+      );
+
+      const { diff } = await this.preCommitProcess({
+        branchName,
+        gitClient,
+        owner,
+        repositoryName,
+      });
+      let fullDiffPath = "";
+      if (diff) {
+        await mkdir(diffFolder, { recursive: true });
+        const diffPath = join(diffFolder, "diff.patch");
+        await writeFile(diffPath, diff);
+        fullDiffPath = resolve(diffPath);
+        console.log("Saving diff to: ", fullDiffPath);
+      }
+
+      await this.provider.createCommit({
+        owner,
+        repositoryName,
+        commitMessage,
+        branchName,
+        files: preparedFiles,
+      });
+
+      if (diff) {
+        await this.postCommitProcess({
+          diffPath: fullDiffPath,
+          gitClient,
         });
+        await rm(fullDiffPath);
+      }
+
+      await rm(cloneDir, { recursive: true, force: true });
+
       const { defaultBranch } = await this.provider.getRepository({
         owner,
         repositoryName,
       });
+
       const existingPullRequest = await this.provider.getPullRequestForBranch({
         owner,
         repositoryName,
         branchName,
       });
-      if (!existingPullRequest) {
-        return this.provider.createPullRequestForBranch({
+
+      let pullRequest: PullRequest | undefined = existingPullRequest;
+
+      if (!pullRequest) {
+        pullRequest = await this.provider.createPullRequestForBranch({
           owner,
           repositoryName,
-          pullRequestTitle,
-          pullRequestBody,
+          pullRequestTitle: accumulativePullRequestTitle,
+          pullRequestBody: accumulativePullRequestBody,
           branchName,
           defaultBranchName: defaultBranch,
         });
       }
-      return existingPullRequest.url;
+
+      await this.provider.commentOnPullRequest({
+        where: { issueNumber: pullRequest.number, owner, repositoryName },
+        data: { body: pullRequestBody },
+      });
+
+      return pullRequest.url;
+    }
+
+    throw new InvalidPullRequestMode();
+  }
+
+  private async preCommitProcess({
+    gitClient,
+    branchName,
+    owner,
+    repositoryName,
+  }: PreCommitProcessArgs): PreCommitProcessResult {
+    console.log("Pre commit process");
```

Comment:
I would replace the console with the new logger. it will give us context and not just empty string